### PR TITLE
Medkit

### PR DESCRIPTION
**Pull request by mgoerwell**
The basic medkit will now heal the player to full health.

### CODE REVIEW

File: src/game/GameStateMatch.cpp [hunk 1]
```diff
@@ -214,7 +214,7 @@ void GameStateMatch::handle() {
                     case SDLK_k:
                         //k is for kill, sets player marine to a nullptr
                         if (GameManager::instance()->getPlayer().getMarine()) {
-                            GameManager::instance()->getPlayer().getMarine()->setHealth(0);
+                            GameManager::instance()->getPlayer().getMarine()->setHealth(50);
```
Comment (isaacmorneau) [inline src/game/GameStateMatch.cpp]: k should still kill the player

File: src/inventory/BasicMedkit.cpp [hunk 2]
```diff
@@ -12,6 +13,7 @@ BasicMedkit::~BasicMedkit() {
     logv("BasicMedkit Destroyed\n");
 }
 
-void BasicMedkit::OnConsume() {
+void BasicMedkit::OnConsume(Marine& x) {
+	x.setHealth(100);
```
Comment (isaacmorneau) [inline src/inventory/BasicMedkit.cpp]: instead of the magic number it should use the health constant in marine

File: src/player/Player.cpp [hunk 3]
```diff
@@ -279,8 +279,8 @@ void Player::handleKeyboardInput(const int winWidth, const int winHeight, const
         marine->inventory.dropWeapon(marine->getX(), marine->getY());
     }
     //use Inventory
-    if (state[SDL_SCANCODE_I]) {
-        marine->inventory.useItem();
+    if(state[SDL_SCANCODE_I]) {
+        marine->inventory.useItem(*marine);
```
Comment (isaacmorneau) [inline src/player/Player.cpp]: this needs to make sure that marine is not nullptr

File: src/inventory/BasicMedkit.cpp [hunk 4]
```diff
@@ -1,17 +1,50 @@
...
+ * Author:      Matthew Goerwell 
+ * Function Interface: OnConsume(Marine& marine)
+ *      Marine &x: A reference to the marine using the medkit.
+ *
+ * Description:
+ *      This is the implementation of the consume method for medkits. It sets the health of the marine
+ *      using it to full health (100). 
+ */
+void BasicMedkit::OnConsume(Marine& marine) {
+	marine.setHealth(100);
```
Comment (isaacmorneau) [inline src/inventory/BasicMedkit.cpp]: this also needs to use the marine constants

**Review by isaacmorneau — CHANGES_REQUESTED** [review]

**Review by JohnAgapeyev — APPROVED** [review]

**Review by isaacmorneau — APPROVED** [review]

**Pull request merged**